core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -109,48 +126,65 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
           else reply ! Committed(offsets.asScala.toMap)
         }
       })
-      //right now we can not store commits in consumer - https://issues.apache.org/jira/browse/KAFKA-3412
-      poll()
+
     case Subscribe(topics, listener) =>
       consumer.subscribe(topics.toList.asJava, new WrappedAutoPausedListener(consumer, listener))
+
     case SubscribePattern(pattern, listener) =>
       consumer.subscribe(Pattern.compile(pattern), new WrappedAutoPausedListener(consumer, listener))
+
     case Poll(target) =>
-      if (target == this) {
-        poll()
-        currentPollTask = schedulePollTask()
-      }
+      if (target == this) poll()
       else {
         // Message was enqueued before a restart - can be ignored
         log.debug("Ignoring Poll message with stale target ref")
       }
+      schedulePollTask()
 
-    case req: RequestMessages =>
+    case sub: SubscriptionMessage =>
       context.watch(sender())
-      checkOverlappingRequests("RequestMessages", sender(), req.topics)
-      requests = requests.updated(sender(), req)
-      poll()
```

Comment:
i was experimenting with solution basing on polls executed only with set interval, lag still was present. Im assuming that in your approach would also, but just guessing. Worth to check